### ISSUE TITLE
Should we use mean rather than sum for batchwise loss reduction?

### ISSUE THREAD

**Issue by jbloomAus** [ai-safety-foundation/sparse_autoencoder#63]
When syncing some changes, I noticed that I went with ".mean()" rather than ".sum()"  in the training pipeline because of [this logic](https://discuss.pytorch.org/t/loss-reduction-sum-vs-mean-when-to-use-each/115641#:~:text=I%20think%20the%20disadvantage%20in%20using%20the%20sum%20reduction%20would%20also%20be%20that%20the%20loss%20scale%20(and%20gradients)%20depend%20on%20the%20batch%20size%2C%20so%20you%20would%20probably%20need%20to%20change%20the%20learning%20rate%20based%20on%20the%20batch%20size.%20While%20this%20is%20surely%20possibly%2C%20a%20mean%20reduction%20would%20not%20make%20this%20necessary.). (you don't want loss magnitude to depend on batch size). 

Relevant code here: https://github.com/jbloomAus/sparse_autoencoder/blob/151a65d58b7379d1fc07ca06cc110fc0647c419d/sparse_autoencoder/train/train_autoencoder.py#L89C1-L93C29

However, maybe there's a reason we want to scale updates with batch size? @alan-cooney

**Comment by alan-cooney** [comment]
No good reason for this!

**Comment by jbloomAus** [comment]
thanks for sorting this!
